### DISCUSSION TITLE
Announcing Floating UI Svelte! 🎉

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/floating-ui-svelte#50] [Announcements]
# Introducing Floating UI Svelte

![Screenshot 2024-05-06 at 7 22 11 PM](https://github.com/skeletonlabs/floating-ui-svelte/assets/1509726/023157d5-3ce0-4253-b571-4fd34505fb46)

Hey everyone, [Chris](https://github.com/endigo9740) here from [Skeleton Labs](https://skeletonlabs.co/). I'm the creator and core maintainer of [Skeleton](https://www.skeleton.dev/), an open source UI toolkit for Svelte. If you're familiar with our project then you know we've been hard at work on our [next major release called Skeleton v3](https://github.com/skeletonlabs/skeleton/discussions/2375).

Alongside this we've been teasing a brand new open source project, and today I'm thrilled to announce [Floating UI Svelte](https://floating-ui-svelte.vercel.app/). If you're not familiar with [Floating UI](https://floating-ui.com/), it's an amazing library for handling positioning and interaction of overlay elements, such as tooltips, drop-down menus, context menus, and more. Skeleton has made use of Floating UI as part of our [Popup feature](https://www.skeleton.dev/utilities/popups) for the last couple years. However, our community has made it clear they [they want a more robust solution](https://github.com/skeletonlabs/skeleton/issues/1916). Today we're answering that with Floating UI Svelte.

Furthermore, Floating UI Svelte is not limited merely to Skeleton users, but for ANYONE in the Svelte community. To put it plainly, if you use Svelte, this is made for you. Whether you're using Skeleton or any other powerful Svelte-related UI library.

![Kapture 2024-05-06 at 20 00 06](https://github.com/skeletonlabs/floating-ui-svelte/assets/1509726/f33ed185-dc67-4e5e-903f-c709c0fc0849)

## 🔗Links

- Video Intro: https://youtu.be/Trr-nn1yZ44
- Website: https://floating-ui-svelte.vercel.app/
- GitHub: https://github.com/skeletonlabs/floating-ui-svelte

### Is the project open source?

Yes, provided under the standard [MIT license](https://github.com/skeletonlabs/floating-ui-svelte/blob/dev/LICENSE).

### Who is maintaining the project?

The project is being built in coordination between [Skeleton Labs](https://skeletonlabs.co/), including contributions from the [Svelte](https://svelte.dev/) and [Skeleton](https://www.skeleton.dev/) open source communities. It is currently maintained by:

- [Hugo Korte](https://github.com/Hugos68) - a long time Svelte and Skeleton contributor
- [Chris Simmons](https://github.com/endigo9740) - the creator and core maintainer for Skeleton

### Do I have to use Skeleton?

No, Floating UI Svelte is built to support any and all Svelte-based projects created using Svelte v5. We will, however, be providing specific guidance and setup instructions for all Skeleton v3 users in the near future.

### Does this support Svelte 5?

Svelte 5 recently [reached release candidate status](https://svelte.dev/blog/svelte-5-release-candidate). As such, we've designed Floating UI Svelte from the ground up to take full advantage of the latest and greatest tools, such as [Runes](https://svelte-5-preview.vercel.app/docs/runes). In fact this is what enables Floating UI Svelte to keep such a familiar API and syntax to the original [Floating UI React](https://floating-ui.com/docs/react) wrapper.

### Why use a library for popovers?

Creating popover interfaces is one of the hardest problems to solve for the modern web. While a native browser [Popover API](https://developer.mozilla.org/en-US/docs/Web/API/Popover_API) is on the horizon, your choices today are fairly limited. In the past you may have solved this using third-party libraries such as [Popper.js](https://popper.js.org/docs/v2/). This is now being replaced by the spiritual successor called [Floating UI](https://floating-ui.com/). This single tool handles all the complex calculations needed to ensure your overlay elements work in even the most complex scenarios. Providing an extensive set of [middleware](https://floating-ui.com/docs/middleware) to easily extend and cover every scenario you can throw at it.

### Why not use Floating UI directly?

Unfortunately Floating UI currently only provides abstractions for [React](https://floating-ui.com/docs/react) and [Vue](https://floating-ui.com/docs/vue). Floating UI Svelte bridges this gap and brings the same quality of life improvements to all Svelte users. Affording you a much more approachable API. While it may initial seem more complex than the current Skeleton offering, you're provided a near limitless set of power in return.

### Is this an official Floating UI port?

No, [but we've expressed interest in collaborating with the maintainers](https://github.com/floating-ui/floating-ui/discussions/2834).

### When can I learn more?

We're still early days, but we will be providing regular updates over the next few weeks. Make sure to keep an eye on our social channels for new announcements.

## 🤝 Contributing

We're actively seeking contributors to help port all hooks, components, and utility features as soon as possible. If you're interested in helping, please reach out in any of the open [GitHub issues](https://github.com/skeletonlabs/floating-ui-svelte/issues), in the comments section below, or via the [Skeleton Discord](https://discord.gg/EXqV7W8MtY)'s `#floating-ui-svelte` channel.

**Comment by baseplate-admin** [comment]
i have a question. Why not add this to `floating-ui` repository? `@floating-ui/vue` is developed inside the mainline repository.

**Reply by endigo9740** [reply]
@baseplate-admin to be clear, Skeleton Labs and the Floating UI team are two separate entities. We did, however, reached out to the Floating UI maintainers and offered to do just that:
https://github.com/floating-ui/floating-ui/discussions/2834

Unfortunately as you can see, they do not have the bandwidth for this. Implementing a whole new framework, new documentation, and all the support that goes along with this is well beyond the scope of a standard contribution. It would require the full attention of the maintainers to pull off.

That said, our preference would be that Floating UI Svelte eventually merge into the official Floating project if/when the maintainers are open to it. And assuming our work meets their standards. But, we've left an open invitation for that!

**Reply by baseplate-admin** [reply]
Alright makes sense to me.

`@floating-ui/vue` was a third party package for a long while before it was merged into mainline repo.

**Reply by endigo9740** [reply]
Oh I didn't realize that. That's great to hear. Perhaps Floating UI Svelte will follow a similar journey then!

**Comment by endigo9740** [comment]
The project is now available in an early access pre-release. Try it yourself and let us know what you think!

- https://github.com/skeletonlabs/floating-ui-svelte/discussions/113
